compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -2674,22 +2674,24 @@ OMR::Z::CodeGenerator::doRegisterAssignment(TR_RegisterKinds kindsToAssign)
          immValue = immValue | (regMask<<8);
          ((TR::S390SILInstruction*)instructionCursor)->setSourceImmediate(immValue);
          }
+         /**
+         * Find a free real register for DCB to use during generate binary encoding phase
+         * @see S390DebugCounterBumpInstruction::generateBinaryEncoding()
+         */
       else if(instructionCursor->getOpCodeValue() == TR::InstOpCode::DCB)
          {
-         TR::S390DebugCounterBumpInstruction *dcbInstr = (TR::S390DebugCounterBumpInstruction *) instructionCursor;
+         TR::S390DebugCounterBumpInstruction *dcbInstr = static_cast<TR::S390DebugCounterBumpInstruction*> instructionCursor;
```

Comment:
You will need parenthesis around the value you are `static_cast`ing otherwise this is a compile time error.